PythonRegression/tests/features/steps/api_test_steps.py

```diff
@@ -41,56 +42,60 @@
 @step(r'"([^"]*)" is called on "([^"]*)" with:')
 def api_method_is_called(step,apiCall,nodeName):
     logger.info('%s is called on %s',apiCall,nodeName)
-    config['apiCall'] = apiCall
-    config['nodeId'] = nodeName
+    world.config['apiCall'] = apiCall
+    world.config['nodeId'] = nodeName
     arg_list = step.hashes
 
     options = {}
     api_utils.prepare_options(arg_list, options)
-    responses[apiCall] = {}
 
     api = api_utils.prepare_api_call(nodeName)
     response = api_utils.fetch_call(apiCall, api, options)
 
     assert type(response) is dict, 'There may be something wrong with the response format: {}'.format(response)
     
-    responses[apiCall] = {}
-    responses[apiCall][nodeName] = response
+    world.responses[apiCall] = {}
+    world.responses[apiCall][nodeName] = response
 
 #This method is identical to the method above, but creates a new thread
 #TODO: Find a way to pass aloe.world variable between threads to eliminate need for duplicate code
 @step(r'"([^"]*)" is called in parallel on "([^"]*)" with:')
 def threaded_call(step,apiCall,node):
     logger.info("Creating thread for {}".format(apiCall))
-    config['apiCall'] = apiCall
-    config['nodeId'] = node
+    world.config['apiCall'] = apiCall
+    world.config['nodeId'] = node
     arg_list = step.hashes
 
     options = {}
     api_utils.prepare_options(arg_list, options)
     api = api_utils.prepare_api_call(node)
 
-    def make_call(api,options):
+    def make_call(api,options,q):
         response = api_utils.fetch_call(apiCall, api, options)
+        responses = q.get()
```

Comment:
looking at it again why do you need a queue?

Reply:
The queue is there to pass the world.responses variable. When trying to access the `world.responses` variable from another thread, you run into a value exception, because the expected variable doesn't exist in that thread.

Reply:
Ok, I assumed that `world` was shared across thread. Apparently it isn't the case